rust/private/toolchain_utils.bzl

```diff
@@ -42,8 +42,6 @@ def _toolchain_files_impl(ctx):
         )
     elif ctx.attr.tool == "rustc_lib":
         files = toolchain.rustc_lib
-    elif ctx.attr.tool == "rustc_srcs":
```

Comment:
just to clarify the effect of this PR: already after this gets submitted, in case a user has a `rust_toolchain` with `rustc_srcs`, they would be ignored by rust_analyzer (even before we drop rust_toolchain.rustc_srcs in a follow-up PR)?

Reply:
This code is actually not quite correct. It does not handle cases where the `rust_toolchain` did not get `rustc_srcs`. Since code here is broken in what I perceive to be the common case I opted to remove it. The toolchain should still support fetching and consuming `rust-src` artifacts but after this change, those sources will not be used by rust-analyzer.

Reply:
thanks